### PR TITLE
Fix false negatives for `isStandardSyntaxDeclaration`

### LINKED ISSUE #3930: Fix false negatives for some rules when using CSS-in-JS parser
> Clearly describe the bug

Some rules don't work with CSS-in-JS parser

> Which rule, if any, is the bug related to?

- `property-no-unknown`
- `value-no-vendor-prefix`
- `max-nesting-depth`

(these are just few random ones I've tested, but probably there are more)

> What CSS is needed to reproduce the bug?

```css
a {
  rule: 1;
}
```

```js
const Component = styled.a`
  rule: 1;
`
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "property-no-unknown": true
  }
}
```

> Which version of stylelint are you using?

`9.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint test.{css,js}`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, `css-in-js`

> What did you expect to happen?

```shell
test.css
 2:3  ×  Unexpected unknown property "rule"   property-no-unknown

test.js
 2:3  ×  Unexpected unknown property "rule"   property-no-unknown
```

> What actually happened (e.g. what warnings or errors did you get)?

```shell
test.css
 2:3  ×  Unexpected unknown property "rule"   property-no-unknown
```

---

This is happening because `postcss-jsx` produces different AST from regular css, which appears to be expected (#3875), making [isStandardSyntaxDeclaration](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxDeclaration.js) return `false` and `property-no-unknown` skipping the declaration.

Aside from core rules, I've seen it in `stylelint-order` plugin, which uses it's own utility function that also not handles this case. I think there would be other plugins like this. If it's impossible to make this parser generate similar AST, I think it would be useful to document it that plugins also should be tested with it's output.

It also would be good to test all core rules for compatibility with that AST.

### PR DESCRIPTION
**Pull request by gucong3000**
> Which issue, if any, is this issue related to?

Closes #3930

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/utils/hasTplInterpolation.js [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
 /**
- * Check whether a string has template literal interpolation
+ * Check whether a string has JS template literal interpolation or HTML-like template
  *
  * @param {string} string
  * @return {boolean} If `true`, a string has template literal interpolation
  */
 module.exports = function(string /*: string*/) /*: boolean*/ {
-  return /\${.+?}/.test(string);
+  return /{.+?}/.test(string);
```
Comment (hudochenkov) [inline lib/utils/hasTplInterpolation.js]: This matches everything.

File: lib/utils/__tests__/isStandardSyntaxUrl.test.js [hunk 2]
```diff
@@ -107,16 +107,16 @@ describe("isStandardSyntaxUrl", () => {
...
-    expect(isStandardSyntaxUrl("@{less-variable}/path")).toBeTruthy();
+    expect(isStandardSyntaxUrl("@{less-variable}/path")).toBeFalsy();
   });
   it("url with less interpolation in the middle without quotes", () => {
-    expect(isStandardSyntaxUrl("some/@{less-variable}/path")).toBeTruthy();
+    expect(isStandardSyntaxUrl("some/@{less-variable}/path")).toBeFalsy();
   });
   it("url with less interpolation at the end without quotes", () => {
-    expect(isStandardSyntaxUrl("some/@{less-variable}")).toBeTruthy();
+    expect(isStandardSyntaxUrl("some/@{less-variable}")).toBeFalsy();
```
Comment (evilebottnawi) [inline lib/utils/__tests__/isStandardSyntaxUrl.test.js]: Looks like a regression

Comment (hudochenkov) [reply lib/utils/__tests__/isStandardSyntaxUrl.test.js]: Actually, it's a bug fix. Interpolation makes URL non-standard. There are tests for Sass interpolation and they all negative.

Comment (evilebottnawi) [reply lib/utils/__tests__/isStandardSyntaxUrl.test.js]: :+1:

**Comment by jeddy3** [comment]
> -   Fixed: false negatives in declaration-based rules for CSS-in-JS ([#3933](https://github.com/stylelint/stylelint/pull/3933)).

**Pull request merged**